TicketsDemo/App_Start/NinjectWebCommon.cs

```diff
@@ -64,20 +66,23 @@ private static IKernel CreateKernel()
         /// <param name="kernel">The kernel.</param>
         private static void RegisterServices(IKernel kernel)
         {
+            kernel.Bind<ICSVPathSettingsService>().To<AppConfigSettingsService>();
+            kernel.Bind<ICSVFieldSettingsService>().To<AppConfigSettingsService>();
             kernel.Bind<ITicketRepository>().To<TicketRepository>();
-            kernel.Bind<ITrainRepository>().To<TrainRepository>();
+            kernel.Bind<ITrainRepository>().To<CSVTrainRepository>();
 
             kernel.Bind<IRunRepository>().To<RunRepository>();
             kernel.Bind<IReservationRepository>().To<ReservationRepository>();
 
             kernel.Bind<ISchedule>().To<Schedule>();
-            kernel.Bind<ITicketService>().To<TicketService>();
+            kernel.Bind<ITicketService>().To<TicketServiceLoggingDecorator>();
+            kernel.Bind<ITicketService>().To<TicketService>().WhenInjectedExactlyInto<TicketServiceLoggingDecorator>();
+            kernel.Bind<ICarriageRepository>().To<CarriageRepository>();
             kernel.Bind<IReservationService>().To<ReservationService>();
-
             //todo factory
-            kernel.Bind<IPriceCalculationStrategy>().To<DefaultPriceCalculationStrategy>();
+            kernel.Bind<IPriceCalculationStrategy>().To<DrinkAndBedCalculationStrategy>();
```

Comment:
DefaultPriceCalculationStrategy is not used but it's code is duplicated within the DrinkAndBedCalculationStrategy. Instead, leave tha main price calculation the way it was done before, then add only new price component calculation logic into the DrinkAndBedCalculationStrategy